Makefile

```diff
@@ -155,20 +153,23 @@ ifeq ($(K3D),true)
 endif
 
 .PHONY: clis
-clis: dist/argo-linux-amd64 dist/argo-linux-ppc64le dist/argo-linux-s390x dist/argo-darwin-amd64 dist/argo-windows-amd64 cli-image
+clis: dist/argo-linux-amd64 dist/argo-linux-arm64 dist/argo-linux-ppc64le dist/argo-linux-s390x dist/argo-darwin-amd64 dist/argo-windows-amd64 cli-image
 
 # controller
 
-dist/workflow-controller-linux-amd64: $(CONTROLLER_PKGS)
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -i -ldflags '${LDFLAGS}' -o dist/workflow-controller-linux-amd64 ./cmd/workflow-controller
+dist/workflow-controller-linux-amd64: GOARGS = GOOS=linux GOARCH=amd64
+dist/workflow-controller-linux-arm64: GOARGS = GOOS=linux GOARCH=arm64
+
+dist/workflow-controller-%: $(CONTROLLER_PKGS)
+	CGO_ENABLED=0 $(GOARGS) go build -v -i -ldflags '${LDFLAGS}' -o $@ ./cmd/workflow-controller
 
 .PHONY: controller-image
 controller-image: dist/controller-image
 
-dist/controller-image: dist/workflow-controller-linux-amd64
+dist/controller-image: dist/workflow-controller-$(GOOS)-$(GOARCH)
```

Comment:
Why to change this?

Reply:
I personally think this change will provide enabling multi-arch image, without harming existing code structure. Reasons given as follows:

On amd64 (x86_64) machines, building target `dist/controller-image` will still give us x86_64 docker images.

On arm64 (aarch64) machines, this change, together with the changes in Dockerfile and Dockerfile.dev, will enable us to build arm64 container images.

After this, if we have above two images, by adding a new docker manifest, we can make our dockehub image multiarch.